pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -126,68 +126,94 @@ type OSDInfo struct {
 }
 
 type OrchestrationStatus struct {
-	OSDs    []OSDInfo `json:"osds"`
-	Status  string    `json:"status"`
-	Message string    `json:"message"`
+	OSDs         []OSDInfo `json:"osds"`
+	Status       string    `json:"status"`
+	PvcBackedOSD bool      `json:"pvc-backed-osd"`
+	Message      string    `json:"message"`
+}
+
+type osdProperties struct {
+	crushHostname  string //it's the hostname if its nodes based OSD provision and the pvc name in case of PVC backed OSD provision
+	devices        []rookalpha.Device
+	pvc            v1.PersistentVolumeClaimVolumeSource
+	selection      rookalpha.Selection
+	resources      v1.ResourceRequirements
+	storeConfig    osdconfig.StoreConfig
+	placement      rookalpha.Placement
+	metadataDevice string
+	location       string
 }
 
 // Start the osd management
 func (c *Cluster) Start() error {
```

Comment:
I wonder if this `Start()` method doesn't need to be modified at all?
- Determining volume sources can be done later when it's time to create the PVCs for the storage class device sets
- The existing code for loading nodes and getting host names should only be relevant for the other type of creating OSDs, which relies on the useAllNodes, useAllDevices, deviceFilter, nodes, and other settings. Why do we need to touch this code when adding device sets? It will just not be triggered for creating OSDs if the settings indicate as such.

Reply:
cc @ashishranjan738

Reply:
ahh, I tried to modularize this part a bit more can you review this part again? cc @travisn